tests/unittests/config/test_schema.py

```diff
@@ -852,14 +1040,127 @@ def test_label_overrides_property_name(self):
+                    "$defs": {"my_ref": {"deprecated": True}},
+                    "properties": {
+                        "prop1": {
+                            "allOf": [
+                                {
+                                    "type": ["string", "integer"],
+                                    "description": "<description>",
+                                },
+                                {"$ref": "#/$defs/my_ref"},
+                            ]
+                        }
+                    },
+                },
+                "**prop1:** (string/integer) DEPRECATED. <description>",
+            ),
+            (
+                {
+                    "$schema": "http://json-schema.org/draft-04/schema#",
+                    "$defs": {
+                        "my_ref": {
+                            "deprecated": True,
+                            "description": "<description>",
+                        }
+                    },
+                    "properties": {
+                        "prop1": {
+                            "allOf": [
+                                {"type": ["string", "integer"]},
+                                {"$ref": "#/$defs/my_ref"},
+                            ]
+                        }
+                    },
+                },
+                "**prop1:** (string/integer) DEPRECATED. <description>",
+            ),
+            (
+                {
+                    "$schema": "http://json-schema.org/draft-04/schema#",
+                    "properties": {
+                        "prop1": {
+                            "description": "<description>",
+                            "anyOf": [
+                                {
+                                    "type": ["string", "integer"],
+                                    "description": "<deprecated_description>",
+                                    "deprecated": True,
+                                },
+                            ],
+                        },
+                    },
+                },
+                "**prop1:** (UNDEFINED) <description>\n",
+            ),
+            (
+                {
+                    "$schema": "http://json-schema.org/draft-04/schema#",
+                    "properties": {
+                        "prop1": {
+                            "anyOf": [
+                                {
+                                    "type": ["string", "integer"],
+                                    "description": "<deprecated_description>",
+                                    "deprecated": True,
+                                },
+                                {
+                                    "type": "number",
+                                    "description": "<description>",
+                                },
+                            ]
+                        },
+                    },
+                },
+                "**prop1:** (UNDEFINED)\n",
```

Comment:
Should this be `(UNDEFINED) <description>\n`?

Reply:
SC-1175